SQHELM-102 prepare the release of the new LTS chart (#289)
SonarSource/helm-chart-sonarqube: charts/sonarqube/README.md

## Introduction

> This chart bootstraps an instance of the latest SonarQube version with a PostgreSQL database. A helm chart containing the SonarQube LTS version can be installed following the steps outlined [here](#installing-the-lts-chart).

Reviewer: Similar suggestion as for sonarqube-dce:
```suggestion
This chart bootstraps an instance of the latest SonarQube version with a PostgreSQL database.

The latest version of the chart installs the latest SonarQube version. To install the version of the chart for SonarQube 9.9 LTS, please read the section [below](#installing-the-lts-chart). Deciding between LTS and Latest? [This may help](https://www.sonarsource.com/products/sonarqube/downloads/lts/)
```

The default login is admin/admin.

> ## Installing the LTS chart

Reviewer: Similar as for sonarqube-dce:
```suggestion
## Installing the SonarQube 9.9 LTS chart
```

Note that the reference to this section would then need to be adjusted in the introduction.

Author: i think you are right, for the sake of the history version, it might be interesting to have it. amending

> The LTS chart is being distributed as the `8.x.x` version of this chart and it is compatible with the current SonarQube LTS (`9.9.x`).

Reviewer: Similar as for sonarqube-dce:
```suggestion
The version of the chart for the SonarQube 9.9 LTS is being distributed as the `8.x.x` version of this chart.
```

> In order to use it, please set the version constraint `~8`, which is equivalent to `>=8.0.0 && <= 9.0.0`. That version parameter **must** be used in every helm related command including `install`, `upgrade`, `template`, and `diff` (don't treat this as an exhaustive list).

Reviewer: Similar as for sonarqube-dce: does it make sense to provide an example?

Author: I think it is worth mentioning.

the helm workflow is so integrated that it is not to be assumed that people knows how the repository update works and how to handle such cases.

Waiting for your opinion on that

Reviewer: I think it's great to mention how to handle the LTS case. My question was in fact a suggestion to add more detailed explanations if we can.

Author: hoo i see, understood it the other way sorry...

That section is like complementary to the `install the chart` section above, but yes we could add a more precise example with something like this ?

```
helm upgrade --install -n sonarqube --version ~8 sonarqube sonarqube/sonarqube
```

Reviewer: 👍

> If you intend to upgrade from the [old and unmaintained LTS chart](https://artifacthub.io/packages/helm/sonarqube/sonarqube-lts) to the newest, please follow the steps [in this section](#upgrade-from-sonarqube-lts-to-this-chart).

Reviewer: > 
```suggestion
To upgrade from the old and unmaintained [sonarqube-lts chart](https://artifacthub.io/packages/helm/sonarqube/sonarqube-lts) to this chart, please follow the steps described [in this section](#upgrade-from-sonarqube-lts-to-this-chart).
```

The default login is admin/admin.

## Installing the SonarQube 9.9 LTS chart

The version of the chart for the SonarQube 9.9 LTS is being distributed as the `8.x.x` version of this chart.

In order to use it, please set the version constraint `~8`, which is equivalent to `>=8.0.0 && <= 9.0.0`. That version parameter **must** be used in every helm related command including `install`, `upgrade`, `template`, and `diff` (don't treat this as an exhaustive list).

Example:
```
helm upgrade --install -n sonarqube --version ~8 sonarqube sonarqube/sonarqube
```

> To upgrade from the old and unmaintained [sonarqube-lts chart](https://artifacthub.io/packages/helm/sonarqube/sonarqube-lts), please follow the steps described [in this section](#upgrade-from-sonarqube-lts-to-this-chart).

Reviewer: The link #upgrade-from-sonarqube-lts-to-this-chart is broken. It might be the case for some others since we changed a few titles in various files.

4. Browse to http://yourSonarQubeServerURL/setup and follow the setup instructions
5. Reanalyze your projects to get fresh data

> ### Upgrade from sonarqube-lts to this chart

Reviewer: ```suggestion
### Upgrade from the old sonarqube-lts chart
```

Note that the links to this section would then need to be adjusted.

For SonarQube 9.9 LTS, the Helm chart version to use is `8.x.x`. 

Before attempting any upgrade, please carefully read the general [upgrade documentation above](#upgrade), the [install-lts guide](#installing-the-lts-chart) and **backup** your db instance. Please also verify that your `values.yaml` file contains only parameters that are still compatible with the targeted chart, adjusting them if needed.

As SonarQube only requires to persist the database, the general upgrade process will consist of uninstalling your instance, before installing the new LTS.

If you are using an external database, you don't have any persistent data inside kubernetes. Therefore, there is no action required. 

Instead, if you rely on the embedded postgres chart (**not recommended**), uninstalling the chart will keep the PVC alive. This can then be reused either:
- by specifying `postgresql.existingClaim` in the `values.yaml` file

> - by not changing parameter values, but making sure you install the new chart in the same namespace (auto-generated name will be the same).

Reviewer: In fact, I realize that we also plan to make it part of SonarQube documentation.
Should we maintain it in both places?

Reviewer: Agree. I think it makes sense to have a single source of truth.
On Docker Hub, we refer upgrade instructions from the documentation. I feel we could do the same here.
